src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -295,222 +392,266 @@ public void start(final CommandMap command)
+
+            // Generate the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countryRawAtlasShardsRDD = getContext()
+                    .parallelize(tasks, tasks.size()).mapToPair(this.generateRawAtlas(boundaries,
+                            sparkContext, atlasLoadingOptions, pbfContext, atlasScheme))
+                    .filter(tuple -> tuple._2() != null);
+
+            // Persist the RDD and save the intermediary state
+            countryRawAtlasShardsRDD.cache();
+            countryRawAtlasShardsRDD.saveAsHadoopFile(
+                    getAlternateSubFolderOutput(output, RAW_ATLAS_FOLDER), Text.class, Atlas.class,
+                    MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE RAW ATLAS **********\n");
+
+            // Slice the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countrySlicedRawAtlasShardsRDD = countryRawAtlasShardsRDD
+                    .mapToPair(this.sliceRawAtlas(boundaries)).filter(tuple -> tuple._2() != null);
+
+            // Persist the RDD and save the intermediary state
+            final String slicedRawAtlasPath = getAlternateSubFolderOutput(output,
+                    SLICED_RAW_ATLAS_FOLDER);
+            countrySlicedRawAtlasShardsRDD.cache();
+            countrySlicedRawAtlasShardsRDD.saveAsHadoopFile(slicedRawAtlasPath, Text.class,
+                    Atlas.class, MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE SLICED RAW ATLAS **********\n");
+
+            // Section the sliced raw Atlas
+            final JavaPairRDD<String, Atlas> countryAtlasShardsRDD = countrySlicedRawAtlasShardsRDD
+                    .mapToPair(this.sectionRawAtlas(boundaries, sharding, sparkContext,
+                            atlasLoadingOptions, slicedRawAtlasPath));
+
+            // Persist the RDD and save the final atlas
+            countryAtlasShardsRDD.cache();
+            countryAtlasShardsRDD.saveAsHadoopFile(
+                    getAlternateSubFolderOutput(output, ATLAS_FOLDER), Text.class, Atlas.class,
+                    MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE FINAL ATLAS **********\n");
+
+            // Create the metrics
+            final JavaPairRDD<String, AtlasStatistics> statisticsRDD = countryAtlasShardsRDD
+                    .mapToPair(generateAtlasStatistics(sharding));
+
+            // Persist the RDD and save
+            statisticsRDD.cache();
+            statisticsRDD.saveAsHadoopFile(
+                    getAlternateSubFolderOutput(output, SHARD_STATISTICS_FOLDER), Text.class,
+                    AtlasStatistics.class, MultipleAtlasStatisticsOutputFormat.class,
+                    new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE SHARD STATISTICS **********\n");
```

Comment:
And again, same here. Although at that point in the job, releasing RDD memory might not make much sense anymore.

Reply:
Actually not here, as it is still used by the Deltas below!